src/core_plugins/kibana/public/dashboard/panel/panel_header/panel_options_menu.js

```diff
@@ -33,56 +35,23 @@ export class PanelOptionsMenu extends React.Component {
 
   onToggleExpandPanel = () => {
     this.closePopover();
-    this.props.onToggleExpandPanel();
+    this.props.toggleExpandedPanel();
   };
 
   renderItems() {
-    const items = [(
-      <KuiContextMenuItem
+    const items = [
```

Comment:
How about the container defines these items too? It's already responsible for partially composing some components, so why not more?

Reply:
Great idea, but I'm a bit stuck because we intercept all actions to add `this.closePopover`.  Can't think of a workaround....

Reply:
I think I figured it out. I tried removing the calls to `closePopover` from all of the callbacks and everything seems OK. From there we can use `componentWillReceiveProps` to let the `isExpanded` prop drive the popover state:

```javascript
  onEditPanel = () => {
    window.location = this.props.editUrl;
  };

  onDeletePanel = () => {
    if (this.props.onDeletePanel) {
      this.props.onDeletePanel();
    }
  };

  onToggleExpandPanel = () => {
    this.props.toggleExpandedPanel();
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.isExpanded !== this.props.isExpanded) {
      this.closePopover();
    }
  }
```

Reply:
As discussed offline, going to leave as is since it feels awkward to rely on isExpanded prop only to close the popover.